ilastik/applets/neuralNetwork/modelStateControl.py

```diff
@@ -0,0 +1,414 @@
+            event.acceptProposedAction()
+
+    def clear(self):
+        super().clear()
+        self.setEnabled(True)
+        self.setTextInteractionFlags(Qt.TextEditorInteraction)
+        self.setToolTip(
+            "Copy/Paste a bioimage.io model doi or nickname, or\ndrag-and-drop a downloaded bioimage.io model.zip file."
+        )
+        self._model_source = None
+        self._initialized = False
+
+    def askClear(self):
+        if (
+            QMessageBox.question(self, "Remove model", "Do you want to remove the model from the project file?")
+            == QMessageBox.Yes
+        ):
+            self.clear()
+            self.modelDeleted.emit()
+
+    def setModelIncompatibleState(self, model_source, model_info, error_message):
+        self.setToolTip(f"Model not compatible with data:\n\n{error_message}")
+        self.setModelInfo(model_source, model_info, template=Template(f"<red>{display_template}</red>"))
+
+    def setEmptyState(self):
+        self.clear()
+
+    def setModelDataAvailableState(self, model_source, model_info):
+        self.btn_container.setEnabled(True)
+        self.setTextInteractionFlags(Qt.NoTextInteraction)
+        self.setToolTip("Remove the model by clicking the 'x' in the upper right corner.")
+        self.setModelInfo(model_source, model_info)
+
+    def setReadyState(self, model_source, model_info):
+        self.btn_container.setEnabled(False)
+        self.setToolTip("Stop the model to make changes.")
+        self.setModelInfo(model_source, model_info)
+
+
+class ModelStateControl(QWidget):
+    uploadDone = pyqtSignal()
+
+    def __init__(self, parent=None):
+        super().__init__(parent)
+        self.threadRouter = ThreadRouter(self)
+        self._preDownloadChecks = set()
+
+        layout = QVBoxLayout()
+
+        self.modelSourceEdit = ModelSourceEdit(self)
+        self.statusLabel = QLabel(self)
+        self.modelControlButton = QToolButton(self)
+        self.modelControlButton.setText("...")
+        self.modelControlButton.setToolTip("Click here to check model details, initialize, or un-initialize the model")
+
+        bottom_layout = QHBoxLayout()
+        bottom_layout.addWidget(self.statusLabel)
+        bottom_layout.addStretch()
+        bottom_layout.addWidget(self.modelControlButton)
+        layout.addWidget(self.modelSourceEdit)
+        layout.addLayout(bottom_layout)
+        self.setLayout(layout)
+
+    def setTiktorchController(self, tiktorchController):
+        self._tiktorchController = tiktorchController
+
+    def setTiktorchModel(self, tiktorchModel):
+        self._tiktorchModel = tiktorchModel
+        self._tiktorchModel.registerListener(self._onTiktorchStateChange)
+        self.modelSourceEdit.modelDeleted.connect(self._tiktorchModel.clear)
+
+    @threadRouted
+    def _onTiktorchStateChange(self, state: TiktorchOperatorModel.State):
+
+        try:
+            self.modelControlButton.clicked.disconnect()
+        except:
```

Comment:
Do we _really_ want to suppress everything, including `BaseException` here? Maybe we want `except Exception`?

```python
with contextlib.suppress(Exception):
        self.modelControlButton.clicked.disconnect()
```